### PR TITLE
Update tensor.rs to Handle Trailing Unit-Dimensions in Image Tensors

### PR DESCRIPTION
**Pull request by tauseefmohammed2**
### What

- Previously, tensor.rs only displayed tensors with shapes of [H, W, 1|3|4] as images, this update addresses the issue by modifying the `image_height_width_channels()`, `is_shaped_like_an_image()`, and `is_vector()` methods to ignore trailing unit-dimensions when checking the shape of the tensor

- This change should improve the user experience by allowing a wider range of image tensor shapes to be displayed as images in tensor.rs

- Resolves Issue: #1871 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)

### CODE REVIEW

File: crates/re_log_types/src/component_types/tensor.rs [hunk 1]
```diff
@@ -424,20 +435,33 @@ impl Tensor {
     }
 
     pub fn is_shaped_like_an_image(&self) -> bool {
```
Comment (emilk) [inline crates/re_log_types/src/component_types/tensor.rs:437]: we can simplify this quite a bit:

```suggestion
    pub fn is_shaped_like_an_image(&self) -> bool {
        self.image_height_width_channels().is_some()
    }
```

File: crates/re_log_types/src/component_types/tensor.rs [hunk 2]
```diff
@@ -424,20 +435,33 @@ impl Tensor {
...
+                1 | 3 | 4
+            )
+        })
     }
 
     #[inline]
     pub fn is_vector(&self) -> bool {
         let shape = &self.shape;
-        shape.len() == 1 || { shape.len() == 2 && (shape[0].size == 1 || shape[1].size == 1) }
+        let mut last_dim = shape.last().unwrap().size;
```
Comment (emilk) [inline crates/re_log_types/src/component_types/tensor.rs:459]: this will panic if `shape` is empty (for "scalar tensors")

**Review by emilk — CHANGES_REQUESTED** [review]

**Comment by emilk** [comment]
Thanks for the PR! It would be great if you added a simple test of this to `examples/python/api_demo/main.py`

**Comment by Wumpf** [comment]
@tauseefmohammed2 you're still on that? :)

**Comment by Wumpf** [comment]
Took the task over in https://github.com/rerun-io/rerun/pull/2075

**Pull request closed without merging**